### ISSUE TITLE
in model.py line 76:   context = attn_weights.bmm(encoder_outputs.transpose(0, 1))  # (B,1,N)

### ISSUE THREAD

**Issue by muzhi-lab-code** [keon/seq2seq#16]
Is this correct? I look up the explanation about bmm in official document,it syas that 
batch1 and batch2 must be 3-D tensors each containing the same number of matrices.
but as you defined before,  the attn_weights is a  2-D shape ,I think here may be some mistakes

**Comment by Hahallo** [comment]
return F.softmax(attn_energies, dim=1).unsqueeze(1)
This line of code makes attn_weights 3-D shape
